tests/staggered_invert_test.cpp

```diff
@@ -318,110 +309,222 @@ void test(int argc, char **argv)
 }
 
 int main(int argc, char **argv)
 {
+  ::testing::InitGoogleTest(&argc, argv);
+  setQudaStaggeredDefaultInvTestParams();
   setQudaDefaultMgTestParams();
   // Parse command line options
   auto app = make_app();
   add_eigen_option_group(app);
   add_deflation_option_group(app);
   add_multigrid_option_group(app);
   add_comms_option_group(app);
-  CLI::TransformPairs<int> test_type_map {{"full", 0}, {"full_ee_prec", 1}, {"full_oo_prec", 2}, {"even", 3},
-                                          {"odd", 4},  {"mcg_even", 5},     {"mcg_odd", 6}};
-  app->add_option("--test", test_type, "Test method")->transform(CLI::CheckedTransformer(test_type_map));
+  add_testing_option_group(app);
+  app->add_option("--legacy-test-info", print_legacy_info, "Print info on how to reproduce the old '--test #' behavior with flags, then exit");
   try {
     app->parse(argc, argv);
   } catch (const CLI::ParseError &e) {
     return app->exit(e);
   }
   setVerbosity(verbosity);
-  if (!inv_multigrid) solve_type = QUDA_INVALID_SOLVE;
-
-  if (inv_deflate && inv_multigrid) {
-    errorQuda("Error: Cannot use both deflation and multigrid preconditioners on top level solve");
-  }
 
   // Set values for precisions via the command line.
   setQudaPrecisions();
 
   // initialize QMP/MPI, QUDA comms grid and RNG (host_utils.cpp)
   initComms(argc, argv, gridsize_from_cmdline);
 
+  if (print_legacy_info) {
+    display_legacy_info();
+    errorQuda("Exiting...");
+  }
+
+  if (inv_deflate && inv_multigrid)
+    errorQuda("Error: Cannot use both deflation and multigrid preconditioners on top level solve.\n");
+
   initRand();
 
-  // Only these fermions are supported in this file. Ensure a reasonable default,
-  // ensure that the default is improved staggered
-  if (dslash_type != QUDA_STAGGERED_DSLASH && dslash_type != QUDA_ASQTAD_DSLASH && dslash_type != QUDA_LAPLACE_DSLASH) {
-    printfQuda("dslash_type %s not supported, defaulting to %s\n", get_dslash_str(dslash_type),
-               get_dslash_str(QUDA_ASQTAD_DSLASH));
-    dslash_type = QUDA_ASQTAD_DSLASH;
+  // Only these fermions are supported in this file
+  if (is_laplace_enabled) {
+    if (!is_staggered(dslash_type) && !is_laplace(dslash_type))
+      errorQuda("dslash_type %s not supported", get_dslash_str(dslash_type));
+  } else {
+    if (is_laplace(dslash_type))
+      errorQuda("The Laplace dslash is not enabled, cmake configure with -DQUDA_LAPLACE=ON");
+    if (!is_staggered(dslash_type))
+      errorQuda("dslash_type %s not supported", get_dslash_str(dslash_type));
   }
 
   // Need to add support for LAPLACE MG?
   if (inv_multigrid) {
-    if (dslash_type != QUDA_STAGGERED_DSLASH && dslash_type != QUDA_ASQTAD_DSLASH) {
-      errorQuda("dslash_type %s not supported for multigrid preconditioner", get_dslash_str(dslash_type));
+    if (!is_staggered(dslash_type)) {
+      errorQuda("dslash_type %s not supported for multigrid preconditioner\n", get_dslash_str(dslash_type));
```

Comment:
No need for `\n` with `errorQuda`

Reply:
https://github.com/lattice/quda/pull/1421/commits/6b0ba62df10a339e90c454a3fc893a3bbddc1985